sui/deploy-contract.js

```diff
@@ -5,156 +5,206 @@ const { toB64 } = require('@mysten/sui/utils');
 const { bcs } = require('@mysten/sui/bcs');
 const { Transaction } = require('@mysten/sui/transactions');
 const {
-    utils: { arrayify, hexlify, toUtf8Bytes, keccak256 },
-    constants: { HashZero },
+    utils: { arrayify },
 } = ethers;
-const { saveConfig, printInfo, validateParameters, writeJSON } = require('../evm/utils');
-const { addBaseOptions } = require('./cli-utils');
+const { saveConfig, printInfo, validateParameters, writeJSON, getDomainSeparator } = require('../common');
+const { addBaseOptions, addDeployOptions, addOptionsToCommands } = require('./cli-utils');
 const { getWallet, printWalletInfo, broadcast } = require('./sign-utils');
-const { loadSuiConfig, getAmplifierSigners, deployPackage, getObjectIdsByObjectTypes } = require('./utils');
 const { bytes32Struct, signersStruct } = require('./types-utils');
-const { upgradePackage } = require('./deploy-utils');
-const { suiPackageAddress, suiClockAddress } = require('./utils');
-
-async function getSigners(keypair, config, chain, options) {
-    if (options.signers === 'wallet') {
-        const pubKey = keypair.getPublicKey().toRawBytes();
-        printInfo('Using wallet pubkey as the signer for the gateway', hexlify(pubKey));
-
-        if (keypair.getKeyScheme() !== 'Secp256k1') {
-            throw new Error('Only Secp256k1 pubkeys are supported by the gateway');
-        }
-
-        return {
-            signers: [{ pub_key: pubKey, weight: 1 }],
-            threshold: 1,
-            nonce: options.nonce ? keccak256(toUtf8Bytes(options.nonce)) : HashZero,
-        };
-    } else if (options.signers) {
-        printInfo('Using provided signers', options.signers);
-
-        const signers = JSON.parse(options.signers);
-        return {
-            signers: signers.signers.map(({ pub_key: pubKey, weight }) => {
-                return { pub_key: arrayify(pubKey), weight };
-            }),
-            threshold: signers.threshold,
-            nonce: arrayify(signers.nonce) || HashZero,
-        };
-    }
-
-    return getAmplifierSigners(config, chain);
+const { upgradePackage, UPGRADE_POLICIES } = require('./deploy-utils');
+const {
+    loadSuiConfig,
+    getSigners,
+    deployPackage,
+    getObjectIdsByObjectTypes,
+    suiPackageAddress,
+    suiClockAddress,
+    readMovePackageName,
+    getChannelId,
+} = require('./utils');
+
+/**
+ * Move Package Directories
+ *
+ * This array contains the names of Move package directories located in:
+ * `node_modules/@axelar-network/axelar-cgp-sui/move`
+ *
+ * Each string in this array corresponds to a folder name within that path.
+ *
+ * To deploy a new package:
+ * 1. Add the new package's folder name to this array
+ * 2. Ensure the corresponding folder exists in the specified path
+ *
+ */
+const PACKAGE_DIRS = ['gas_service', 'test', 'axelar_gateway'];
```

Comment:
read this dynamically by going over all dirs instead? we do something kind of similar for EVM contracts
https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/evm/utils.js#L811

Reply:
Yeah, i've thought about it. I've set it manually because I'd like to avoid having additional checks in the program flows to ensure we've already handled that contract deployment, otherwise we throw error early. 

So i think it's cleaner to define supported packages at the beginning, outside of the program flow. Then, dynamically generated the sub-commands instead based on that values instead. The main idea of this pattern will also align with your suggestion to define a map to handle deployment script, instead of switch-case.